### PR TITLE
fix(backup): enqueue volumes with correct names

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#5411, longhorn/longhorn#10057

#### What this PR does / why we need it:

In `enqueueVolumesForBackupVolume` of volume controller, it needs to use BackupVolume.Spec.VolumeName` to get volume CR instead of `BackupVolume.Name`.

#### Additional documentation or context
Run test cases:
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7981
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7986 
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7995
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/8001
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/8012
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/8013
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/8014
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/8015

### CODE REVIEW

**Comment by mantissahz** [comment]
@Mergifyio backport v1.8.x

File: controller/system_backup_controller.go [hunk 1]
```diff
@@ -834,7 +834,7 @@ func (c *SystemBackupController) WaitForVolumeBackupToComplete(backups map[strin
 		for name := range backups {
 			// Retrieve the latest backup
 			var backup *longhorn.Backup
-			backup, err = c.ds.GetBackup(name)
+			backup, err = c.ds.GetBackupRO(name)
```
Comment (derekbit) [inline controller/system_backup_controller.go:837]: What's the purpose of the change?

File: controller/system_backup_controller.go [hunk 2]
```diff
@@ -864,6 +867,21 @@ func (c *SystemBackupController) WaitForVolumeBackupToComplete(backups map[strin
 	return fmt.Errorf("unexpected error: stopped waiting for Volume backups without completing, failing or timing out")
 }
 
+func (c *SystemBackupController) isVolumeLastBackupSynced(backup *longhorn.Backup) bool {
+	snapshot, err := c.ds.GetSnapshotRO(backup.Status.SnapshotName)
+	if err != nil {
+		c.logger.WithError(err).Warnf("Failed to get snapshot %v for backup %v", backup.Status.SnapshotName, backup.Name)
+		return false
+	}
+	volume, err := c.ds.GetVolume(snapshot.Spec.Volume)
```
Comment (derekbit) [inline controller/system_backup_controller.go:876]: Can we use `GetVolumeRO`?

Comment (mantissahz) [reply controller/system_backup_controller.go]: Modified.Thanks

File: controller/system_backup_controller.go (hunk 1 above)
Comment (mantissahz) [reply controller/system_backup_controller.go:837]: We only read `backup` and will not modify it so use `GetBackupRO` here.

File: controller/system_backup_controller_test.go [hunk 3]
```diff
@@ -371,3 +377,35 @@ func fakeSystemRolloutNamespace(c *C, informerFactory informers.SharedInformerFa
 	err = nsIndexer.Add(namespace)
 	c.Assert(err, IsNil)
 }
+
+func fakeSystemRolloutSnapshot(fakeObj *longhorn.Snapshot, c *C, informerFactory lhinformers.SharedInformerFactory, client *lhfake.Clientset) {
```
Comment (c3y1huang) [inline controller/system_backup_controller_test.go:381]: Move this to `system_restore_controller_test.go`.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: controller/system_backup_controller_test.go (hunk 3 above)
Comment (mantissahz) [reply controller/system_backup_controller_test.go]: Sure, and moved. Thanks

**Review by c3y1huang — APPROVED** [review]

**Review by derekbit — APPROVED** [review]

**Comment by derekbit** [comment]
@mergify backport v1.8.x
